### ISSUE TITLE
main wrapping element

### ISSUE THREAD

**Issue by mattgi** [yabwe/medium-editor#251]
By default, the editor will use `<p>` tags to contain content and support carriage returns, etc. Alt., the `disableReturn` option can be used to force plain text, thereby removing all wrapping `<p>` tags.

I have a requirement to support formatting inside block elements like Headers and Blockquotes, but I need to also support carriage returns and do not want `<p>` tags to be used.

Standard:

```
<div class="editor">
  <p>this is the <i>first</i> line</p>
  <p>this is a <b>second</b> line</p>
</div>
```

Requirement:

```
<h1 class="editor">
  this is the <i>first</i> line<br />this is a <b>second</b> line
</h1>
```

Perhaps an option like `disableParagraphReturn: true` is what I am looking for - where it would use `<br>`'s instead of `<p>`'s when set to `true`.

Another option may be to simply consider the element that has been used for the editor - where it exhibits a behavior without an option being passed (e.g. `<div>` means that `<p>` tags will be used; `<h1>`..`<h6>`, `<blockquote>` will use `<br>`'s).

I will start to investigate how this may be implemented if it is a feature that others may want.

**Comment by thomstark** [comment]
I want! I want all <p> tags and all <divs> disabled, and only <br> tags when hitting Return.

**Comment by yoshokatana** [comment]
This would also be useful if the containing element were itself a paragraph, e.g.

```
<p class="editor">
  this is the <i>first</i> line<br />this is a <b>second</b> line
</p>
```

...which is sort of how I'm using it right now. (It still tries to insert extraneous `<p>` tags, see https://github.com/yabwe/medium-editor/issues/34#issuecomment-123733549 )

**Comment by alvarotrigo** [comment]
Very interested on seeing this behavior too.

**Comment by alvarotrigo** [comment]
Interesting... using `display:inline-block` forces `<br>` on `enter` press.
Not a medium-editor issue but a contenteditable one. I found it out [here](http://stackoverflow.com/a/20533364/1081396).

[Reproduction online](https://jsfiddle.net/ay9cLgab/20/)

**Comment by alvarotrigo** [comment]
This behaviour is totally different in Firefox. It seems Firefox doesn't even create new `p` elements.

**Comment by zetura** [comment]
@alvarotrigo Thanks for the tip!
And it's not blocking the creation of LI elements (with the code from medium.js demo) if you put a UL/OL around.
I needed to be able to edit headers and paragraphs without creating new ones, but also lists and creating new <li> elements in it.

**Comment by dypsilon** [comment]
I am also interested in this feature and would love to see some progress here.

**Comment by mscreenie** [comment]
Same. I don't want wrapping of anything, maybe just standard <br> for new lines.

**Comment by volumetric** [comment]
Anyone found a solution for this?

**Comment by skiokko** [comment]
I also agree that have <br> for new lines by default is better. Generate new <p> for every line break create problems.

**Comment by cmans** [comment]
In the medium-editor.js file, around like 6822, or search this:
`        // https://github.com/yabwe/medium-editor/issues/994
        // Firefox thrown an error when calling `formatBlock` on an empty editable blockContainer that's not a <div>
        if (MediumEditor.util.isMediumEditorElement(node) && node.children.length === 0 && !MediumEditor.util.isBlockContainer(node)) {
            this.options.ownerDocument.execCommand('formatBlock', false, 'p');
        }`

That p should be blank. That will remove the wrapping p tags.

**Comment by alvarotrigo** [comment]
Thanks ghost, apparently firefox fixed it:
https://github.com/yabwe/medium-editor/issues/994

**Comment by DavidGeismarLtd** [comment]
Any updates on this ? How do you get brs instead of ps ?
